Python/optimizer_analysis.c

```diff
@@ -482,7 +364,6 @@ sym_set_type(_Py_UOpsSymType *sym, _Py_UOpsSymExprTypeEnum typ, uint64_t refinem
     }
 }
 
-// Note: for this, to_sym MUST point to brand new sym.
 static void
 sym_copy_immutable_type_info(_Py_UOpsSymType *from_sym, _Py_UOpsSymType *to_sym)
```

Comment:
This could use a comment explaining why it's doing what it does. I still don't understand why.